### PR TITLE
🛖 Creating new workspaces (child and separate)

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: templates/spa_frame.haml [hunk 1]
```diff
@@ -862,6 +869,8 @@
                 -for other_org in user_orgs
                   -if other_org.id != user_org.id
                     %temba-option(name='{{ other_org.name }}' value="{{other_org.id}}")
+                -if org_perms.orgs.org_create_new and user_org.is_multi_org and not user_org.parent
```
Comment (norkans7) [inline templates/spa_frame.haml:872]: Monosnap RapidPro - Visually build nationally scalable mobile applications 2022-11-18 14-49-39

Monosnap RapidPro - Visually build nationally scalable mobile applications 2022-11-18 14-49-53

File: templates/frame.haml [hunk 2]
```diff
@@ -195,8 +195,11 @@
                 .org.text-primary
                   %a.posterize{href:'{% url "orgs.org_choose"%}?organization={{org.pk}}'}
                     {{ org.name }}
-
-
+            -if org_perms.orgs.org_create_new and user_org.is_multi_org
```
Comment (norkans7) [inline templates/frame.haml:198]: Monosnap RapidPro - Visually build nationally scalable mobile applications 2022-11-18 14-49-39

Monosnap RapidPro - Visually build nationally scalable mobile applications 2022-11-18 14-49-53

File: templates/spa_frame.haml (hunk 1 above)
Comment (norkans7) [inline templates/spa_frame.haml:872]: Monosnap Inbox 2022-11-18 14-50-09

Monosnap Create Org 2022-11-18 14-50-25

File: templates/frame.haml (hunk 2 above)
Comment (norkans7) [reply templates/frame.haml:198]: The link to create a new workspace will properly be hidden when workspace has `is_multi_org=False`

File: templates/spa_frame.haml (hunk 1 above)
Comment (norkans7) [reply templates/spa_frame.haml:872]: No easy way to hide the New workspace option for workspace without permission when the workspace is switched how the view will redirect to workspace home

Comment (norkans7) [reply templates/spa_frame.haml:872]: Any ideas on how to improve this? @ericnewcomer

**Comment by rowanseymour** [comment]
Captura de pantalla 2022-11-24 a la(s) 10 23 45

Captura de pantalla 2022-11-24 a la(s) 15 01 20

File: temba/orgs/models.py [hunk 3]
```diff
@@ -517,28 +517,32 @@ def get_unique_slug(cls, name):
 
             return unique_slug
 
-    def create_child(self, user, name: str, timezone, date_format: str):
```
Comment (rowanseymour) [inline temba/orgs/models.py:520]: I convinced myself that in the interests of simplicity, you don't typically need to specify date format when creating a new org.

File: temba/orgs/tests.py [hunk 4]
```diff
@@ -2862,7 +2868,7 @@ def test_home(self):
 
         self.login(self.admin)
 
-        with self.assertNumQueries(45):
+        with self.assertNumQueries(46):
```
Comment (rowanseymour) [inline temba/orgs/tests.py:2871]: now we always check if there are child orgs to know if you should be able to acces that page.. it seems to me that even if you lose access to the child_orgs feature, you should still be able to see existing child orgs.

File: temba/tests/crudl.py [hunk 5]
```diff
@@ -47,7 +47,7 @@ def as_user(user, allowed):
             else:
                 checks = [LoginRedirectOr404()]
 
-            return self.requestView(url, user, checks=checks)
+            return self.requestView(url, user, checks=checks, choose_org=self.org)
```
Comment (rowanseymour) [inline temba/tests/crudl.py:50]: to make these methods work when there are multiple orgs, we now always choose `self.org`

File: templates/frame.haml [hunk 6]
```diff
@@ -182,9 +182,8 @@
       <!-- Header -->
       -if user_org
         .org-header.self-end
-          -if user_orgs|length > 1
-            .pull-right.toggle
-              .icon-menu-2
+          .pull-right.toggle
+            .icon-menu-2
```
Comment (rowanseymour) [inline templates/frame.haml:186]: menu now always visible

File: templates/frame.haml [hunk 7]
```diff
@@ -195,8 +194,11 @@
                 .org.text-primary
                   %a.posterize{href:'{% url "orgs.org_choose"%}?organization={{org.pk}}'}
                     {{ org.name }}
-
-
+            -if org_perms.orgs.org_create
```
Comment (norkans7) [inline templates/frame.haml:197]: What about the new UI for this?

Comment (rowanseymour) [reply templates/frame.haml:197]: @ericnewcomer will take care of that

File: temba/orgs/views.py [hunk 8]
```diff
@@ -2621,45 +2619,71 @@ def get_context_data(self, **kwargs):
...
-        def derive_initial(self):
-            initial = super().derive_initial()
-            initial["timezone"] = self.request.org.timezone
-            initial["date_format"] = self.request.org.date_format
-            return initial
+        def pre_process(self, request, *args, **kwargs):
+            # if org has neither feature then redirect
+            features = self.request.org.features
+            if Org.FEATURE_NEW_ORGS not in features and Org.FEATURE_CHILD_ORGS not in features:
+                return HttpResponseRedirect(reverse("users.user_login"))
```
[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/orgs/views.py:2648]: yeah.. ugh the reason I did it this was that I convinced myself it doesn't matter where you redirect to, and the CRUDLTestMixin methods all assume that if org2 can't access a page, it's a login redirect...

[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/orgs/views.py]: updated

**Pull request merged**